### ISSUE TITLE
Bug: [no-invalid-void-type] `Promise<void[]>` being flagged

### ISSUE THREAD

**Issue by SukkaW** [typescript-eslint/typescript-eslint#9463]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=BQQwzgngdgxgBMAlHAvAPjgbwFBzjAeyjABc4AHAJwIFsBLMAUzAC44AFa%2BpgHgDcCdACZoA2gF1UcCQG5sufEVIUAjG060GjfoKESMKDly0A6EABtzwKpqZhEcvIWJlyAJnXHeA4QaO3GM0trL2YHbABfRCQZIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDcBDRgBNaPDpWFFS6KImjQO0SODABfECqA&tsconfig=&tokens=false

### Repro Code

```TypeScript
(async () => {
  const promises: Promise<void>[] = [];

  const p1: Promise<void[]> = Promise.all(promises);
  const p2: Promise<void> = Promise.all(promises);
})();
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-invalid-void-type": "error"
  }
}
```

### tsconfig

```jsonc
{}
```

### Expected Result

`Promise<void[]>` is not flagged.

### Actual Result

`Promise<void[]>` is flagged.

**Comment by bradzacher** [comment]
See https://github.com/typescript-eslint/typescript-eslint/issues/8113
